src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/DSA/DSASignVerify.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Dsa.Tests
+{
+    public class DSASignVerify
+    {
+        [Fact] //todo: 2048 not always supported
+        public static void InvalidKeySize_DoesNotInvalidateKey()
+        {
+            using (DSA dsa = DSAFactory.Create())
+            {
+                byte[] signature = dsa.SignData(DSATestData.HelloBytes, HashAlgorithmName.SHA1);
+
+                // A 2049-bit key is hard to describe, none of the providers support it.
+                Assert.ThrowsAny<CryptographicException>(() => dsa.KeySize = 2049);
+
+                Assert.True(dsa.VerifyData(DSATestData.HelloBytes, signature, HashAlgorithmName.SHA1));
+            }
+        }
+
+        [Fact]
+        public static void SignAndVerifyDataNew1024()
+        {
+            using (DSA dsa = DSAFactory.Create(1024))
+            {
+                //SignAndVerify(DSATestData.HelloBytes, "SHA1", dsa.ExportParameters(true));
+                byte[] signature = dsa.SignData(DSATestData.HelloBytes, new HashAlgorithmName("SHA1"));
+                //Assert.Equal(48, signature.Length);
+                bool signatureMatched = dsa.VerifyData(DSATestData.HelloBytes, signature, new HashAlgorithmName("SHA1"));
+                Assert.True(signatureMatched);
+            }
+        }
+
+        [Fact]
+        public static void SignAndVerifyDataExplicit1024()
+        {
+            SignAndVerify(DSATestData.HelloBytes, "SHA1", DSATestData.GetDSA1024Params());
+        }
+
+        [Fact] //todo: 2048 not always supported
+        public static void SignAndVerifyDataExplicit2048()
+        {
+            SignAndVerify(DSATestData.HelloBytes, "SHA256", DSATestData.GetDSA2048Params());
+        }
+
+        [Fact]
+        public static void VerifyKnownSignature()
+        {
+            // The parameters and signature come from FIPS 186-2 APPENDIX 5. EXAMPLE OF THE DSA
```

Comment:
Took this from desktop tests. There are tests that I didn't port, many cng-specific some of which I may add to the Cng assembly.